server/engine/engine.go

```diff
@@ -153,7 +153,26 @@ func (engine *Engine) startDeploymentExecutions() {
 	}
 }
 
+func (engine *Engine) SetFinalStatus() {
```

Comment:
The name of the function suggests that it is setting a final status but it is not. Let's give it a name that describes exactly what the function does, for example "ReportFinalDeploymentStatusToMetrics" or "ReportFinalDeploymentStatusToTelemetry".

Reply:
@bhavenst , do you agree with this naming as well? Want to avoid another review asking for name change.

Reply:
@bhavenst What do you think about the suggested name change?

Reply:
I'm not one for nitpicking naming, but it is just a one/two line change to hopefully make it a bit clearer in the future and then we can merge it immediately so why not..

Reply:
ok, done.